realm/src/androidTest/java/io/realm/RealmListTest.java

```diff
@@ -47,6 +49,143 @@ protected void tearDown() throws Exception {
         testRealm.close();
     }
 
+    public void testPublicNoArgConstructor() {
+        RealmList<AllTypes> list = new RealmList<AllTypes>();
+        assertNotNull(list);
+    }
+
+    public void testUnavailableMethodsInNonManagedMode() {
+        RealmList<AllTypes> list = new RealmList<AllTypes>();
+        try {
+            list.move(0, 1);
+            fail("move() should fail in non-managed mode.");
+        } catch (RealmException ignore) {
+        }
+        try {
+            list.where();
+            fail("where() should fail in non-managed mode.");
+        } catch (RealmException ignore) {
+        }
+    }
+
+    public void testAddNonManagedMode() {
+        RealmList list = new RealmList();
+        AllTypes object = new AllTypes();
+        object.setColumnString("String");
+        list.add(object);
+        assertEquals(1, list.size());
+        assertEquals(object, list.get(0));
+    }
+
+    public void testAddNullNonManagedMode() {
+        RealmList list = new RealmList();
+        try {
+            list.add(null);
+            fail("Adding null should not be be allowed");
+        } catch (NullPointerException ignore) {
+        }
+    }
+
+    public void testAddManagedObject_nonManagedMode() {
+        RealmList list = new RealmList();
+        testRealm.beginTransaction();
+        AllTypes managedAllTypes =  testRealm.createObject(AllTypes.class);
+        testRealm.commitTransaction();
+        try {
+            list.add(managedAllTypes);
+            fail("Adding managed objects to non-managed lists should fail");
+        } catch (IllegalStateException ignore) {
+        }
+    }
+
+    public void testAddAtIndex_nonManagedMode() {
+        RealmList list = new RealmList();
+        AllTypes object = new AllTypes();
+        object.setColumnString("String");
+        list.add(0, object);
+        assertEquals(1, list.size());
+        assertEquals(object, list.get(0));
+    }
+
+    public void testAddManagedObjectAtIndex_nonManagedMode() {
+        RealmList list = new RealmList();
+        testRealm.beginTransaction();
+        AllTypes managedAllTypes = testRealm.createObject(AllTypes.class);
+        testRealm.commitTransaction();
+        try {
+            list.add(5, managedAllTypes);
+            fail("Adding managed objects to non-managed lists should fail");
+        } catch (IllegalStateException ignore) {
+        }
+    }
+
+    public void testAddNullAtIndex_nonManagedMode() {
+        RealmList list = new RealmList();
+        try {
+            list.add(null);
+            fail("Adding null should not be be allowed");
+        } catch (NullPointerException ignore) {
```

Comment:
Isn't NullPointerException wrong to use here? Shouldn't it be IllegalArgumentException (similar other places)